### PR TITLE
[improve][broker] Make some operation subscription dispatchRate methods in Namespaces async

### PR DESCRIPTION
**Pull request by gaoyanliang**
### Motivation

See PIP https://github.com/apache/pulsar/issues/14365 and change tracker https://github.com/apache/pulsar/issues/15043.

Make setSubscriptionDispatchRate / getSubscriptionDispatchRate / deleteSubscriptionDispatchRate methods to pure async.

### Documentation

- [x] `no-need-doc`

### CODE REVIEW

**Comment by gaoyanliang** [comment]
@shibd @Technoboy- @mattisonchao  PTAL

**Comment by gaoyanliang** [comment]
Hi @Technoboy- Can you help trigger the ci ？

**Comment by shibd** [comment]
@gaoyanliang CI should be able to trigger automatically because you are not submitting pr for the first time.

When the ci job failed, you can use cmd`/pulsarbot run-failure-checks` to rerun these jobs(write in the comment area).

**Comment by gaoyanliang** [comment]
> @gaoyanliang CI should be able to trigger automatically because you are not submitting pr for the first time.
> 
> When the ci job failed, you can use cmd`/pulsarbot run-failure-checks` to rerun these jobs(write in the comment area).

ok, get it. Thanks

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java [hunk 1]
```diff
@@ -772,7 +772,7 @@ public void deleteDispatchRate(@PathParam("tenant") String tenant, @PathParam("n
     @ApiResponses(value = { @ApiResponse(code = 403, message = "Don't have admin permission"),
             @ApiResponse(code = 404, message = "Namespace does not exist") })
     public DispatchRate getDispatchRate(@PathParam("tenant") String tenant,
-            @PathParam("namespace") String namespace) {
+                                        @PathParam("namespace") String namespace) {
```
Comment (shibd) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java:775]: invalid change

**Review by shibd — APPROVED** [review]
/LGTM

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java [hunk 2]
```diff
@@ -849,12 +849,20 @@ public DispatchRate getDispatchRate(@PathParam("property") String property, @Pat
...
+                                            @PathParam("property") String property,
                                             @PathParam("cluster") String cluster,
                                             @PathParam("namespace") String namespace,
                                             DispatchRateImpl dispatchRate) {
         validateNamespaceName(property, cluster, namespace);
-        internalSetSubscriptionDispatchRate(dispatchRate);
+        internalSetSubscriptionDispatchRateAsync(dispatchRate)
+                .thenAccept(__ -> asyncResponse.resume(Response.noContent().build()))
+                .exceptionally(ex -> {
+                    log.error("Failed to set the subscription dispatchRate for cluster on namespace {}",
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java:861]: We add clientAppId in the log like :l
log.error("[{}] Failed to set the subscription dispatchRate for cluster on namespace {}", clientAppId(), namespaceName, ex);

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java [hunk 3]
```diff
@@ -877,11 +893,19 @@ public DispatchRate getSubscriptionDispatchRate(@PathParam("property") String pr
...
+                                               @PathParam("property") String property,
                                                @PathParam("cluster") String cluster,
                                                @PathParam("namespace") String namespace) {
         validateNamespaceName(property, cluster, namespace);
-        internalDeleteSubscriptionDispatchRate();
+        internalDeleteSubscriptionDispatchRateAsync()
+                .thenAccept(__ -> asyncResponse.resume(Response.noContent().build()))
+                .exceptionally(ex -> {
+                    log.error("Failed to delete the subscription dispatchRate for cluster on namespace {}"
+                            , namespaceName, ex);
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java:905]: and here

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java [hunk 4]
```diff
@@ -797,12 +797,21 @@ public DispatchRate getDispatchRate(@PathParam("tenant") String tenant,
...
+                                            @ApiParam(value =
+                                            "Subscription dispatch rate for all topics of the specified namespace")
                                                         DispatchRateImpl dispatchRate) {
         validateNamespaceName(tenant, namespace);
-        internalSetSubscriptionDispatchRate(dispatchRate);
+        internalSetSubscriptionDispatchRateAsync(dispatchRate)
+                .thenAccept(__ -> asyncResponse.resume(Response.noContent().build()))
+                .exceptionally(ex -> {
+                    log.error("Failed to update the subscription dispatchRate for cluster on namespace {}"
+                            , namespaceName, ex);
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java:811]: and here

**Comment by gaoyanliang** [comment]
@Technoboy-  @mattisonchao  ptal.

**Comment by gaoyanliang** [comment]
@Technoboy-  @mattisonchao  All checks have passed. PTAL

**Comment by gaoyanliang** [comment]
@Technoboy- @mattisonchao  PTAL。

**Review by Technoboy- — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 5]
```diff
@@ -1385,46 +1385,32 @@ protected CompletableFuture<DispatchRate> internalGetTopicDispatchRateAsync() {
...
+                    policies.subscriptionDispatchRate.remove(pulsar().getConfiguration().getClusterName());
+                    log.info("[{}] Successfully delete the subscriptionDispatchRate for cluster on namespace {}",
+                            clientAppId(), namespaceName);
+                    return policies;
+                }));
     }
 
-    protected DispatchRate internalGetSubscriptionDispatchRate() {
+    protected CompletableFuture<DispatchRate> internalGetSubscriptionDispatchRateAsync() {
         validateNamespacePolicyOperation(namespaceName, PolicyName.RATE, PolicyOperation.READ);
```
Comment (RobertIndie) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1409]: Need to remove this line.

**Review by RobertIndie — CHANGES_REQUESTED** [review]

**Review by Demogorgon314 — APPROVED** [review]

**Review by RobertIndie — APPROVED** [review]

**Review by mattisonchao — APPROVED** [review]

**Pull request merged**